GSS-Cogs/idpd-frontend-datasets
# Logging library

This is the same ticket as this one for the [homepage](https://github.com/GSS-Cogs/idpd-frontend-homepage/pull/56).

This ticket was to improve how our logger works.

Using [next-logger](https://github.com/sainsburys-tech/next-logger) to output logs in json, which uses [pino](https://github.com/pinojs/pino) to format and add fields to the output. Using the next-logger.config to configure our logs to conform to [dp logging standards](https://github.com/ONSdigital/dp-standards/blob/main/LOGGING_STANDARDS.md).
It's setup so full needed log is only done in production and then we use [pino-pretty](https://github.com/pinojs/pino-pretty) in dev so we have more readable outputs in the terminal.

to test
do yarn dev go on the app, should see some nice looking logs in terminal.
Add some of your own console.info and console.error to see different outputs.

do make build and make start you should now see the output in json format in terminal.
